api/v1alpha3/virtualmachine_types.go

```diff
@@ -357,6 +361,65 @@ type VirtualMachineCdromSpec struct {
 	AllowGuestControl *bool `json:"allowGuestControl,omitempty"`
 }
 
+// VirtualMachineCryptoSpec defines the desired state of a VirtualMachine's
+// encryption state.
+type VirtualMachineCryptoSpec struct {
+	// +optional
+
+	// ClassName describes the name of the EncryptionClass resource
+	// used to encrypt this VM.
+	//
+	// Please note, this field is not required to encrypt the VM. If the
+	// underlying platform has a default key provider, the VM may still be fully
+	// or partially encrypted depending on the specified storage and VM classes.
+	//
+	// If there is a default key provider and an encryption storage class is
+	// selected, the VM's home files and non-PVC disks will be encrypted.
+	//
+	// If there is a default key provider and a and a VM Class with a virtual,
+	// trusted platform module (vTPM) is selected, the VM's home files will be
+	// encrypted.
+	//
+	// If the underlying vSphere platform does not have a default key provider,
+	// then this field is required when specifying an encryption storage class
+	// and/or a VM Class with a vTPM.
+	ClassName string `json:"className,omitempty"`
```

Comment:
What are your thoughts on calling this `encryptionClassName`?  There's no stuttering since the parent field is called `spec.crypto`.   IMHO, `spec.cryto.encryptionClassName` is more accurate than `spec.crypto.className`.

Not to mention we have another field called `className` which might be confusing.

Reply:
Let me think about it. I am leaning towards agreeing with you.

Reply:
Done, thanks @aruneshpa !